### ISSUE TITLE
``.. uml::`` directive also used by plantUML

### ISSUE THREAD

**Issue by AlbertMietus** [alendit/sphinx-pyreverse#4]
This extension has a hardcode directive to trigger the extension: `` .. uml::``. Which is used by (at least) the plantUML extension also.
 (As that one specify UML-design/diagrams there, that use is more appropriate)

Plz change this, e.g by using the directive ``.. pyreverse::``
And/or make it configurable (from conf.py)

Thanks

_Note: I would like to use this (or similar) extension. However it has some issues. I  will report them one-by-one._

**Comment by doublethefish** [comment]
We will likely close this as invalid, although add a config option may be a good idea.

**Comment by AlbertMietus** [comment]
Plz don’t close, without fixing/changing.

Especially, now pyreversy  in Sphinx becomes beter maintained (see #6) the combination of plantUML — in the  design phase— and pyreversy — in the integration  phase— is very convenient

By having other directives, this becomes possible without tricks

Thanks

**Comment by alendit** [comment]
`.. uml::` might be indeed to generic. I'd have nothing against `.. pyreverse::` or `.. pyreverse-uml::`. @doublethefish thoughts?

**Comment by doublethefish** [comment]
I have spent a lot of time think about this, and there are two important considerations:

1. If we change the directive from ``uml`` then we break back compatibility. That is damaging to sphinx-pyreverse.
2. If plantUML is using ``uml`` for their directive why can't they change....

Either way I have implemented a workaround that allows configuration of the directive used by sphinx-pyreverse via an env-variable. See PR #15 . Unfortunately, I don't see any way to achieve this using conf.py but willing to listen to ideas.

One option that might help with point (1) above is to implement some deprecation strategy, but that would require some investment of time and thinking.
